stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/list/CustomEventHandler.kt

```diff
@@ -0,0 +1,76 @@
+package io.getstream.chat.ui.sample.feature.channel.list
+
+import io.getstream.chat.android.client.api.models.FilterObject
+import io.getstream.chat.android.client.events.ChannelUpdatedByUserEvent
+import io.getstream.chat.android.client.events.ChannelUpdatedEvent
+import io.getstream.chat.android.client.events.MemberAddedEvent
+import io.getstream.chat.android.client.events.MemberRemovedEvent
+import io.getstream.chat.android.client.events.NotificationAddedToChannelEvent
+import io.getstream.chat.android.client.events.NotificationMessageNewEvent
+import io.getstream.chat.android.client.events.NotificationRemovedFromChannelEvent
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.offline.querychannels.BaseChatEventHandler
+import io.getstream.chat.android.offline.querychannels.ChatEventHandlerFactory
+import io.getstream.chat.android.offline.querychannels.EventHandlingResult
+import io.getstream.chat.ui.sample.common.isDraft
+import kotlinx.coroutines.flow.StateFlow
+
+class CustomChatEventHandlerFactory : ChatEventHandlerFactory() {
+    override fun chatEventHandler(channels: StateFlow<List<Channel>>) = CustomChatEventHandler(channels)
+}
+
```

Comment:
If it's kind of refence implementation it would be great to have some KDOCs that gives you some overview of the idea.

Reply:
It's inside Sample app and I would say it won't be a common scenario so I would rather skip that for now